server/src/test/java/org/opensearch/cluster/NodeConnectionsServiceTests.java

```diff
@@ -490,6 +492,72 @@ public void testDebugLogging() throws IllegalAccessException {
         }
     }
 
+    public void testConnectionCheckerRetriesIfPendingDisconnection() throws InterruptedException {
+        final Settings.Builder settings = Settings.builder();
+        final long reconnectIntervalMillis = 50;
+        settings.put(CLUSTER_NODE_RECONNECT_INTERVAL_SETTING.getKey(), reconnectIntervalMillis + "ms");
+
+        final DeterministicTaskQueue deterministicTaskQueue = new DeterministicTaskQueue(
+            builder().put(NODE_NAME_SETTING.getKey(), "node").build(),
+            random()
+        );
+
+        MockTransport transport = new MockTransport(deterministicTaskQueue.getThreadPool());
+        TestTransportService transportService = new TestTransportService(transport, deterministicTaskQueue.getThreadPool());
+        transportService.start();
+        transportService.acceptIncomingRequests();
+
+        final NodeConnectionsService service = new NodeConnectionsService(
+            settings.build(),
+            deterministicTaskQueue.getThreadPool(),
+            transportService
+        );
+        service.start();
+
+        // setup the connections
+        final DiscoveryNode node = new DiscoveryNode("node0", buildNewFakeTransportAddress(), Version.CURRENT);
+        ;
+        final DiscoveryNodes nodes = DiscoveryNodes.builder().add(node).build();
+
+        final AtomicBoolean connectionCompleted = new AtomicBoolean();
+        service.connectToNodes(nodes, () -> connectionCompleted.set(true));
+        deterministicTaskQueue.runAllRunnableTasks();
+        assertTrue(connectionCompleted.get());
+
+        // now trigger a disconnect, and then set pending disconnections to true to fail any new connections
+        final long maxDisconnectionTime = 1000;
+        final long disconnectionTime = 100;
+        deterministicTaskQueue.scheduleAt(disconnectionTime, new Runnable() {
```

Comment:
Not sure the intent of running this at fixed schedule ?

Reply:
agree this is not necessary, we can schedule immediately

Reply:
changed to immediately schedule